source/guide/3.2/guide-security.md

```diff
@@ -11,33 +11,230 @@ pageord: 500
+details and verify the given credentials indeed match.<br/>
+To support a variety of user-store systems and configurations Cloudify security framework can accept different
+userstore implementations. It's possible to use the default Flask-secuREST simple userstore or to specify a new
+implementation that supports a specific userstore system.
+
+A valid userstore implementation can be any Python class that inherits from
+[abstract_userstore.py](https://github.com/cloudify-cosmo/flask-securest/blob/master/flask_securest/userstores/
+abstract_userstore.py) and implements a `get_user` method.
+The object returned by `get_user` must adhere to Flask-secuREST's User Model, specifying these 3 methods:
+
+- is_active()
+- is_anonymous()
+- get_roles()
+
+
+## Authentication Provider
+An Authentication Provider is a class that performs authentication. Multiple authentication providers can be configured
+in order to support different authentication methods (e.g. password, token, Kerberos).<br/>
+When a REST call is received by the REST service, the security framework will attempt to authenticate it using each
+authentication provider in the order of configuration, until authentication succeeds.
+The authentication provider has access to the userstore instance (if configured) and can use it to get user details and
+use them to perform authentication.<br/>
+For example, it can compare the given password to the one found on the userstore or verify the user is still active
+(in many environments users are marked as "inactive", instead of deleting the account entirely).
+
+
+Once an authenticator can successfully authenticate the request's user - it should return the user object and allow the
+request to be completed. Other authenticators will not be called until the next request is processed.
+If non of the authenticators authenticates the request - the request will not reach its endpoint and the client will
+receive an "Unauthorized User" error.
+
+A valid authentication provider implementation can be any Python class that inherits from
+[abstract_authentication_provider.py](https://github.com/cloudify-cosmo/flask-securest/blob/master/flask_securest/
+authentication_providers/abstract_authentication_provider.py) and implements an `authenticate` method.
+
+
+## Token Generator
+We mentioned Token as an authentication method. But in order to send a token with each request, the user must first
+receive a token. Tokens can be generated by many systems, and they will work as long as the token can be processed by
+one of the registered authentication providers.
+
+
+To make things easier, Cloudify can also generate tokens through the REST service endpoint "/tokens".
+To enable this feature a token generator must be configured.
+{%note title=Note%}
+The request to "MANAGER_IP/tokens" must itself be authenticated (using a username-password set, for example).
```

Comment:
"MANAGER_IP/tokens" --> "/tokens" endpoint
more consistent with what you've been going for so far